### PR TITLE
Add DSL docs and update bootstrap project

### PR DESCRIPTION
**Pull request by dan-blanchard**
I think this is good enough for a preview release now.

Rendered docs available on RTD [here](http://streamparse.readthedocs.org/en/feature-dsl_docs/)

### CODE REVIEW

**Comment by dan-blanchard** [comment]
I believe I've finally got all the relevant DSL things updated for the preview release.  You can see this branch on RTD (if you have the direct link) [here](http://streamparse.readthedocs.org/en/feature-dsl_docs/).

**Comment by dan-blanchard** [comment]
Bah, forgot the stupid quickstart instructions.  Doing those now.

**Comment by dan-blanchard** [comment]
Okay, now it's gone.

**Comment by rduplain** [comment]
Add method link to `emit_direct` in `streamparse.Grouping` API docs:

```rst
:meth:`~...emit_direct`
```

File: streamparse/storm/bolt.py [hunk 1]
```diff
@@ -12,6 +12,50 @@ class JavaBolt(Component):
...
+
+        :param name:   Name of this Bolt.  Defaults to name of
+                       :class:`~streamparse.Topology` attribute this is assigned
+                       to.
+        :type name:    `str`
+        :param serialized_java:  Serialized Java code representing the class.
+                                 You must either specify this, or
+                                 both ``full_class_name`` and ``args_list``.
+        :type serialized_java: `bytes`
+        :param full_class_name: Full name of Java class, including namespaces.
```
Comment (rduplain) [inline streamparse/storm/bolt.py]: I suppose that "including namespaces" makes sense to a Python developer, but a Java developer would think of it as "fully qualified class name, i.e. include the package name".

File: streamparse/storm/spout.py [hunk 2]
```diff
@@ -12,16 +12,79 @@ class JavaSpout(Component):
...
+
+        :param name:   Name of this Spout.  Defaults to name of
+                       :class:`~streamparse.Topology` attribute this is assigned
+                       to.
+        :type name:    `str`
+        :param serialized_java:  Serialized Java code representing the class.
+                                 You must either specify this, or
+                                 both ``full_class_name`` and ``args_list``.
+        :type serialized_java: `bytes`
+        :param full_class_name: Full name of Java class, including namespaces.
```
Comment (rduplain) [inline streamparse/storm/spout.py]: Same as other comment.

**Comment by rduplain** [comment]
This looks great. I added some minor comments. Once addressed, +1 to merge.

**Pull request merged**